src/main/java/fi/aalto/cs/intellij/common/Resources.java

```diff
@@ -0,0 +1,71 @@
+package fi.aalto.cs.intellij.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class that provides methods to read the resource files of the plugin application.
+ * Normally, this class should be accessed via {@code Resources.SingletonUtility.INSTANCE}.
+ */
+class Resources {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(Resources.class);
+
+  private ResourceProvider resourceProvider;
+
+  Resources(@NotNull ClassLoader classLoader) {
+    this(classLoader::getResourceAsStream);
+  }
+
+  Resources(@NotNull ResourceProvider resourceProvider) {
+    this.resourceProvider = resourceProvider;
+  }
+
+  /**
+   * Returns properties from a properties resource file.
+   * @param resourceName Name of a resource.
+   * @return Contents of the resource in a {@link Properties} object.
+   */
+  @Nullable
+  public Properties getProperties(@NotNull String resourceName) {
+    try (InputStream stream = resourceProvider.getResourceAsStream(resourceName)) {
+      return getProperties(stream);
```

Comment:
this eventually works and `try` seems to be able to take parameters (I'm kinda surprised :open_mouth:), thus, for the sake of simplicity I suggest to use more conventional approach like:

`   
    try {
      InputStream stream = resourceProvider.getResourceAsStream(resourceName);
      return getProperties(stream);
    } catch (Exception ex) {
`

https://docs.oracle.com/javase/tutorial/essential/exceptions/try.html

Reply:
I didn't know about that kind of `try` clauses either. First I wrote:
```
    InputStream stream = null;
    try {
      stream = resourceProvider.getResourceAsStream(resourceName);
      return getProperties(stream);
    } catch (Exception ex) {
      LOGGER.error("Could not read a resource: " + resourceName, ex);
      return null;
    }
    finally {
      if (stream != null) {
        try {
          stream.close();
        } catch (IOException e) { }
      }
    }
```
but then IntelliJ said I should use that simpler form. `Alt+Shift+Enter` and - boom! - I got the committed code.
![image](https://user-images.githubusercontent.com/6066364/72790525-7235f880-3c3e-11ea-9973-240395cf90bf.png)